### PR TITLE
Stream settings

### PR DESCRIPTION
**Pull request by CasperMok**
* Alphanumeric sort by source ID
* Stream settings in both the stream and single view
* Settings button

### CODE REVIEW

File: LocalPackages/DolbyIORTSUIKit/Sources/DolbyIORTSUIKit/Private/ViewModels/StreamViewModel.swift [hunk 1]
```diff
@@ -105,7 +105,9 @@ final class StreamViewModel: ObservableObject {
             sortedSources = sources
         case .alphaNumeric:
             sortedSources = sources.sorted {
-                $0.streamId.localizedStandardCompare($1.streamId) == .orderedAscending
+                let a = $0.sourceId.value ?? " "
+                let b = $1.sourceId.value ?? " "
+                return a.localizedStandardCompare(b) == .orderedAscending
```
Comment (aravind-raveendran) [inline LocalPackages/DolbyIORTSUIKit/Sources/DolbyIORTSUIKit/Private/ViewModels/StreamViewModel.swift:110]: If we want to define - how sorting is done between two different sources we can use something like below:

extension StreamSource: Comparable {
    public static func < (lhs: StreamSource, rhs: StreamSource) -> Bool {
        switch (lhs.sourceId, rhs.sourceId) {
        case (.main, .main), (.main, .other):
            return false
        case (.other, .main):
            return true
        case let (.other(lhsSourceId), .other(rhsSourceId)):
            return lhsSourceId < rhsSourceId
        }
    }
}

File: LocalPackages/DolbyIORTSUIKit/Sources/DolbyIORTSCore/Model/StreamSource.swift [hunk 2]
```diff
@@ -20,7 +20,7 @@ public struct StreamSource: Equatable, Hashable, Identifiable {
             }
         }
 
-        var value: String? {
+        public var value: String? {
```
Comment (aravind-raveendran) [inline LocalPackages/DolbyIORTSUIKit/Sources/DolbyIORTSCore/Model/StreamSource.swift:23]: Do we need `value` public ?

File: LocalPackages/DolbyIORTSUIKit/Sources/DolbyIORTSUIKit/Components/StreamingScreen.swift [hunk 3]
```diff
@@ -10,9 +10,15 @@ public struct StreamingScreen: View {
...
+    @State private var isShowingSettingsScreen: Bool = false
+    @State private var streamId: String?
 
-    public init(isShowingStreamView: Binding<Bool>) {
+    private let settingsManager: SettingsManager
+
+    public init(isShowingStreamView: Binding<Bool>,
+                settingManager: SettingsManager = .shared) {
         _isShowingStreamView = isShowingStreamView
+        self.settingsManager = settingManager
```
Comment (aravind-raveendran) [inline LocalPackages/DolbyIORTSUIKit/Sources/DolbyIORTSUIKit/Components/StreamingScreen.swift:21]: Is settings manager needed in View ? Can be accessed from the ViewModels instead ? Same for file `SingleStreamView.swift`

File: LocalPackages/DolbyIORTSUIKit/Sources/DolbyIORTSUIKit/Components/StreamingScreen.swift [hunk 4]
```diff
@@ -32,6 +38,12 @@ public struct StreamingScreen: View {
             }
             .hidden()
 
+            NavigationLink(destination: LazyNavigationDestinationView(SettingsScreen()),
+                           isActive: $isShowingSettingsScreen
```
Comment (aravind-raveendran) [inline LocalPackages/DolbyIORTSUIKit/Sources/DolbyIORTSUIKit/Components/StreamingScreen.swift:42]: intendation is a bit off here ?

File: LocalPackages/DolbyIORTSUIKit/Sources/DolbyIORTSCore/Model/StreamSource.swift (hunk 2 above)
Comment (CasperMok) [reply LocalPackages/DolbyIORTSUIKit/Sources/DolbyIORTSCore/Model/StreamSource.swift:23]: No if we put the compare in an extention

File: LocalPackages/DolbyIORTSUIKit/Sources/DolbyIORTSUIKit/Components/StreamingScreen.swift (hunk 4 above)
Comment (CasperMok) [reply LocalPackages/DolbyIORTSUIKit/Sources/DolbyIORTSUIKit/Components/StreamingScreen.swift:42]: will do

File: LocalPackages/DolbyIORTSUIKit/Sources/DolbyIORTSUIKit/Private/Views/SettingsButton.swift [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+//
+
+import DolbyIOUIKit
+import SwiftUI
+
+struct SettingsButton: View {
+
+    @Binding private var isShowingSettingsScreen: Bool
+
+    init(isShowingSettingsScreen: Binding<Bool>) {
```
Comment (aravind-raveendran) [inline LocalPackages/DolbyIORTSUIKit/Sources/DolbyIORTSUIKit/Private/Views/SettingsButton.swift:12]: a better way would be to just pass the completion closure in as an argument, as this binding is not used in this View

**Review by aravind-raveendran — APPROVED** [review]

File: LocalPackages/DolbyIORTSUIKit/Sources/DolbyIORTSUIKit/Private/Views/SettingsButton.swift (hunk 5 above)
Comment (CasperMok) [reply LocalPackages/DolbyIORTSUIKit/Sources/DolbyIORTSUIKit/Private/Views/SettingsButton.swift:12]: May do it in the next PR

**Pull request merged**